components/input-validation-mgt/org.wso2.carbon.identity.input.validation.mgt/src/main/java/org/wso2/carbon/identity/input/validation/mgt/listener/InputValidationListener.java

```diff
@@ -0,0 +1,138 @@
+import org.wso2.carbon.identity.input.validation.mgt.services.InputValidationManagementServiceImpl;
+import org.wso2.carbon.identity.mgt.policy.PolicyViolationException;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.ERROR_CODE_PREFIX;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.Configs.PASSWORD;
+import static org.wso2.carbon.identity.input.validation.mgt.utils.Constants.ErrorMessages.ERROR_NO_CONFIGURATIONS_FOUND;
+
+/**
+ * Lister class to validate the password.
+ */
+public class InputValidationListener extends AbstractIdentityUserOperationEventListener {
+
+    private static final Log LOG = LogFactory.getLog(InputValidationListener.class);
+    private final InputValidationManagementService inputValidationMgtService =
+            new InputValidationManagementServiceImpl();
+
+    @Override
+    public int getExecutionOrderId() {
+
+        int orderId = getOrderId();
+        if (orderId != IdentityCoreConstants.EVENT_LISTENER_ORDER_ID) {
+            return orderId;
+        }
+        return 3;
+    }
+
+    public boolean doPreAddUser(String userName, Object credential, String[] roleList, Map<String, String> claims,
+                                String profile, UserStoreManager userStoreManager) throws UserStoreException {
+
+        if (UserCoreUtil.getSkipPasswordPatternValidationThreadLocal()) {
+            return true;
+        }
+        return validate(PASSWORD, credential.toString(), userStoreManager);
+    }
+
+    public boolean doPreUpdateCredentialByAdminWithID(String userID, Object newCredential,
+                                                      UserStoreManager userStoreManager) throws UserStoreException {
+
+        return validate(PASSWORD, newCredential.toString(), userStoreManager);
+    }
+
+    /**
+     * Method to validate password.
+     *
+     * @param field             Credential of the user.
+     * @param value             Value to be validated.
+     * @param userStoreManager  User store manager.
+     * @return  Validity of the password.
+     * @throws UserStoreException   If an error occurred while validating password.
+     */
+    private boolean validate(String field, String value, UserStoreManager userStoreManager) throws UserStoreException {
+
+        int tenantId = userStoreManager.getTenantId();
+        String tenantDomain = IdentityTenantUtil.getTenantDomain(tenantId);
+
+        List<ValidationConfiguration> configurations;
+        try {
+            configurations = inputValidationMgtService.getInputValidationConfiguration(tenantDomain);
+            configurations = configurations.stream().filter(f -> field.equalsIgnoreCase(f.getField()))
```

Comment:
shall we rename `f` variable to a meaningful name?

Reply:
Addressed with 064a761984c1eb2d8cd3d04da73cec40c62f6bbc